airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_web_analytics_streams.py

```diff
@@ -0,0 +1,567 @@
+        object_type,
+        parent_stream_associations,
+        http_mocker: HttpMocker
+    ):
+        self.mock_oauth(http_mocker, self.ACCESS_TOKEN)
+        self.mock_scopes(http_mocker, self.ACCESS_TOKEN, [])
+        self.read_from_stream(self.oauth_config(), stream_name, SyncMode.full_refresh, expecting_exception=True)
+
+    @pytest.mark.parametrize(("stream_name", "parent_stream_name", "object_type", "parent_stream_associations"), CRM_STREAMS)
+    @HttpMocker()
+    def test_given_unauthorized_error_when_read_then_hault(
+        self,
+        stream_name,
+        parent_stream_name,
+        object_type,
+        parent_stream_associations,
+        http_mocker: HttpMocker
+    ):
+        self.mock_custom_objects(http_mocker)
+        self.mock_properties(http_mocker, object_type, self.PROPERTIES)
+        self.mock_parent_object(
+            http_mocker, [self.OBJECT_ID], object_type, parent_stream_name, parent_stream_associations, list(self.PROPERTIES.keys())
+        )
+        self.mock_response(
+            http_mocker,
+            self.web_analytics_request(stream_name, self.ACCESS_TOKEN, self.OBJECT_ID, object_type),
+            HttpResponse(status_code=http.HTTPStatus.UNAUTHORIZED, body="{}")
+        )
+        with mock.patch("time.sleep"):
+            output = self.read_from_stream(self.private_token_config(self.ACCESS_TOKEN), stream_name, SyncMode.full_refresh)
+        assert len(output.records) == 0
+        assert len(output.trace_messages) > 0
+        assert len(output.errors) > 0
+
+    @pytest.mark.parametrize(("stream_name", "parent_stream_name", "object_type", "parent_stream_associations"), CRM_STREAMS)
+    @HttpMocker()
+    def test_given_one_page_when_read_then_get_transformed_records(
+        self,
+        stream_name,
+        parent_stream_name,
+        object_type,
+        parent_stream_associations,
+        http_mocker: HttpMocker
+    ):
+        self.mock_custom_objects(http_mocker)
+        self.mock_properties(http_mocker, object_type, self.PROPERTIES)
+        self.mock_parent_object(
+            http_mocker, [self.OBJECT_ID], object_type, parent_stream_name, parent_stream_associations, list(self.PROPERTIES.keys())
+        )
+        self.mock_response(
+            http_mocker,
+            self.web_analytics_request(stream_name, self.ACCESS_TOKEN, self.OBJECT_ID, object_type),
+            self.web_analytics_response(stream_name)
+        )
+        output = self.read_from_stream(self.private_token_config(self.ACCESS_TOKEN), stream_name, SyncMode.full_refresh)
+        record = output.records[0].record.data
+        assert "properties" not in record
+        prop_fields = len([f for f in record if f.startswith("properties_")])
+        assert prop_fields > 0
+
+    @pytest.mark.parametrize(("stream_name", "parent_stream_name", "object_type", "parent_stream_associations"), CRM_STREAMS)
+    @HttpMocker()
+    def test_given_one_page_when_read_then_get_no_records_filtered(
```

Comment:
Why would they be filtered?

Reply:
we have a [filter](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py#L330) in one of the base classes that can be turned on or off per class, so this test is checking that this specific stream does not filter out the records